### PR TITLE
Improve debug info of some tests

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
From recent flakefinder we see these tests to fail. Lets improve the debug info to help us identify root cause.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by brianmcarey — COMMENTED** [review]
/lgtm

File: tests/infrastructure/tls-configuration.go [hunk 1]
```diff
@@ -49,38 +49,35 @@ import (
 
 var _ = DescribeInfra("tls configuration", func() {
 
-	var (
-		virtClient kubecli.KubevirtClient
-		cipher     *tls.CipherSuite
-	)
+	var virtClient kubecli.KubevirtClient
```
Comment (EdDev) [inline tests/infrastructure/tls-configuration.go:52]: nit: There is no need for this, just call it multiple times, it is a singleton.

File: tests/infrastructure/tls-configuration.go [hunk 2]
```diff
@@ -102,10 +99,10 @@ var _ = DescribeInfra("tls configuration", func() {
 					CipherSuites:       kvtls.CipherSuiteIds([]string{cipher.Name}),
 				}
 				conn, err := tls.Dial("tcp", fmt.Sprintf("localhost:844%d", i), acceptedTLSConfig)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(conn).ToNot(BeNil())
-				Expect(conn.ConnectionState().Version).To(BeEquivalentTo(tls.VersionTLS12))
-				Expect(conn.ConnectionState().CipherSuite).To(BeEquivalentTo(cipher.ID))
+				Expect(conn).ToNot(BeNil(), fmt.Sprintf("Should accept valid tls config, %s", err))
+				Expect(err).ToNot(HaveOccurred(), "Should accept valid tls config")
```
Comment (EdDev) [inline tests/infrastructure/tls-configuration.go:103]: One liner is enough here (error is asserted implicitly):
```
Expect(tls.Dial("tcp", fmt.Sprintf("localhost:844%d", i), acceptedTLSConfig)).ToNot(BeNil(), fmt.Sprintf("Should accept valid tls config, %s", err))
```

File: tests/infrastructure/tls-configuration.go [hunk 3]
```diff
@@ -49,38 +49,35 @@ import (
...
-		labelSelectorList := []string{"kubevirt.io=virt-api", "kubevirt.io=virt-handler", "kubevirt.io=virt-exportproxy"}
+		By("Verify Kubevirt has correct configuration")
+		newKv := util.GetCurrentKv(virtClient)
+		Expect(newKv.Spec.Configuration.TLSConfiguration.MinTLSVersion).To(BeEquivalentTo(v1.VersionTLS12))
+		Expect(newKv.Spec.Configuration.TLSConfiguration.Ciphers).To(BeEquivalentTo([]string{cipher.Name}))
 
+		labelSelectorList := []string{"kubevirt.io=virt-api", "kubevirt.io=virt-handler"}
+		if checks.HasFeature(virtconfig.VMExportGate) {
+			labelSelectorList = append(labelSelectorList, "kubevirt.io=virt-exportproxy")
+		}
```
Comment (EdDev) [inline tests/infrastructure/tls-configuration.go:80]: Why did you moved this to the test itself? All these steps seem to be a pre-requisite of this test, not part of the test itself. It is often useful to separate the two, knowing if it is a test setup issue or the functionality issue.

**Comment by alicefr** [comment]
The code looks good
/approve
/hold 
Put on hold to address Ed's comment

File: tests/infrastructure/tls-configuration.go (hunk 3 above)
Comment (xpivarc) [reply tests/infrastructure/tls-configuration.go:80]: I moved this so the test can run in any configuration, with or without the feature gate. I expect us to use labels for skipping not the skip we have right now.

File: tests/infrastructure/tls-configuration.go (hunk 2 above)
Comment (xpivarc) [reply tests/infrastructure/tls-configuration.go:103]: Do you suggest that checking the TLS version and used cipher is redundant?

File: tests/infrastructure/tls-configuration.go (hunk 1 above)
Comment (xpivarc) [reply tests/infrastructure/tls-configuration.go:52]: I would prefer to tackle this pattern globally.

File: tests/infrastructure/tls-configuration.go (hunk 3 above)
Comment (xpivarc) [reply tests/infrastructure/tls-configuration.go:80]: Well, I actually wanted to drop the KV check, seems redundant, wdyt?

File: tests/infrastructure/tls-configuration.go (hunk 2 above)
Comment (xpivarc) [reply tests/infrastructure/tls-configuration.go:103]: I see but in this case, I also need the `conn` to assert the tls&cipher. Therefore I need to leave the call as is and can't use it inside the `Expect`. In theory, I could write a transformer function but that will not make things more readable in this case imho.

File: tests/infrastructure/tls-configuration.go (hunk 3 above)
Comment (xpivarc) [reply tests/infrastructure/tls-configuration.go:80]: I agree, I will drop it for now.

**Comment by xpivarc** [comment]
/hold cancel
@EdDev PTAL

**Review by EdDev — APPROVED** [review]
Thanks!

**Pull request merged**